keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+the developer focused NetworkPolicy API in Kubernetes.
+
+## Motivation
+
+Kubernetes provides the NetworkPolicy resource to control traffic within a
+cluster. NetworkPolicy focuses on expressing a developer's intent to secure
+their applications. However, it was not intended to be used for cluster scoped
+administrative traffic control, which is reflected by its design:
+- NetworkPolicy uses a "implicit isolation" model, which means that once a policy
+  is applied to certain workloads, they are automatically isolated (in the direction
+  specified by the policy) and anything allowed needs to be explicitly called out.
+  It has no concept of explicit "deny" rules, because the application deployer can
+  simply refrain from allowing the things they want to deny.
+- It doesn't include priorities, or any concept of one user "outranking" another
+  and being able to override their policies, because a given application is expected
+  to only have policies created by a single user (the one deploying the application).
+Thus, in order to satisfy the needs of a cluster admin, we propose to introduce
+a new API that captures the administrator's intent.
+
+### Goals
+
+The goals for this KEP are to satisfy the following key user stories:
+
+1. As a cluster administrator, I want to enforce irrevocable guardrails that
+   all workloads must adhere to in order to guarantee the safety of my clusters.
+   In particular I want to enforce certain network level access controls that are
+   cluster scoped and cannot be overridden or bypassed by namespace scoped
+   NetworkPolicies.
+
+   Example: I would like to explicitly allow all pods in my cluster to reach
+   kubeDNS.
+
+2. As a cluster administrator, I want to have the option to enforce network level
+   access controls that facilitate network multi-tenancy and strict network level
+   isolation between multiple teams and tenants sharing a cluster via use of namespaces
+   or groupings of namespaces per tenant.
+
+   Example: I would like to define two tenants in my cluster, one composed of the pods
+   in `foo-ns-1` and `foo-ns-2` and the other with pods in `bar-ns-1`, where inter-tenant
+   traffic is denied.
+
+3. As a cluster administrator, I want to optionally also deploy an additional default
+   set of policies to all workloads that may be overridden by the developers if needed
+
+   Example: I would like to explicitly delegate the restriction of traffic destined
+   for cluster monitoring pods to the developer, allowing them to setup network policy
+   to deny or allow the traffic from/to their application.
+
+There are several unique properties that we need to add in order accomplish the
+user stories above.
+1. Deny rules and, therefore, hierarchical enforcement of policy
+2. Semantics for a cluster-scoped policy object that may include
+   namespaces/workloads that have not been created yet.
+3. Interoperability with existing Kubernetes Network Policy API
```

Comment:
We should provide e2e tests for this, I've added a comment in the Test Plan section

Reply:
Done